torch/package/_mangling.py

```diff
@@ -0,0 +1,52 @@
+"""Import mangling.
+See mangling.md for details.
+"""
+import re
+
+_mangle_index = 0
+
+
+class PackageMangler:
+    """
+    Used on import, to ensure that all modules imported have a shared mangle parent.
+    """
+
+    def __init__(self):
+        global _mangle_index
+        self._mangle_index = _mangle_index
+        # Increment the global index
+        _mangle_index += 1
+        # Angle brackets are used so that there is almost no chance of
+        # confusing this module for a real module. Plus, it is Python's
+        # preferred way of denoting special modules.
+        self._mangle_parent = f"<torch_package_{self._mangle_index}>"
+
+    def mangle(self, name) -> str:
+        return self._mangle_parent + "." + name
+
+    def demangle(self, mangled) -> str:
```

Comment:
I am not sure this version is really needed and which one of the demangles to use might confuse people working with this code. I think it is safe to always use the global demangle since the only case where we use this one is `_get_package` where we know already that the name is mangled locally.